tests/system/src/test/java/test/robot/javafx/scene/TooltipTest.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.robot.javafx.scene;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Tooltip;
+import javafx.scene.input.MouseButton;
+import javafx.scene.layout.StackPane;
+import javafx.scene.robot.Robot;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+import javafx.stage.Window;
+import javafx.util.Duration;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import test.util.Util;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class TooltipTest {
+    static CountDownLatch startupLatch = new CountDownLatch(1);
+    static CountDownLatch tooltipShownLatch;
+    static Robot robot;
+    static Button button;
+    static Tooltip tooltip;
+
+    static volatile Stage stage;
+    static volatile Scene scene;
+
+    private static void assertTooltipShowDelay(long tooltipShowTime, long expectedTime) {
+        assertTooltipShowDelay(tooltipShowTime, expectedTime, 50);
```

Comment:
If I change this to 100, then all tests pass most of the time. So maybe 150 or 200 would be a better maximum delta?

If you do make this change, then you will need to find all of the three-arg calls with a maxDifference less than that and change them to the two-arg version.

Reply:
even with max=200 we might see occasional failures due to load or other circumstances.

I wonder a better approach would be to check whether the measurement of the delay falls sufficiently outside of the default 1000 ms (which in itself might be more than that)?

Reply:
We have many tests that will occasionally fail with a heavily loaded system.

The question is whether we can come up with a max delta that we can reliably use that is less than the difference between the default value and the test value. Otherwise, we can't distinguish them.

On two different test systems, both of which fail pretty consistently with  50 msec, I see a 100% pass rate over several tries with 150 msec. More testing is needed.

Reply:
Another possibility is the code that measures the time it takes to show the tooltip.

The current code uses Util.waitForLatch(), L244 which was written for a different use case and actually introduces a small measurement error as it includes the time needed for the context switch.  Perhaps a better solution would be to note the timestamp in L271 where the tooltip listener gets called (similar to the way the start moment is captured in L241)

Reply:
Good catch. Capturing the current time in the listener will be more accurate than doing it after the latch wait.

Reply:
Good catch indeed. Will test and report back accordingly.